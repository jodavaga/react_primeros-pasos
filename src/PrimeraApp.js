import React from 'react';
import PropTypes from 'prop-types';

// Functional component

const PrimeraApp = ({ saludo, subtitulo }) => {

    return(
        <>
            <h1>{ saludo }</h1>
            <small>{subtitulo }</small>
        </>    
    )
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    subtitulo: PropTypes.string
}

PrimeraApp.defaultProps = {
    subtitulo: 'Subtitulo by default prop'
}

export {
    PrimeraApp as default
}