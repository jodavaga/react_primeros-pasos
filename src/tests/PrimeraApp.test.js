import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
// import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';

describe('Primera App tests', () => {

    // test('Should display saludo: "Hola, soy jose"', () => {
    //     const mensaje = 'Hola, soy jose';

    //     const {getByText} = render( <PrimeraApp saludo={ mensaje } /> );

    //     expect(getByText(mensaje)).toBeInTheDocument();
    // });

    test('Should render <PrimeraApp /> correctly', () => {
        
        const mensaje = 'Hola, soy jose';
        const wrapper = shallow( <PrimeraApp saludo={mensaje} /> )

        expect( wrapper ).toMatchSnapshot();
    });

    test('Should display subtitle sent by props', () => {
        
        const mensaje = 'Hola, soy jose';
        const subtitle = 'Subtitulo sent by props';
        const wrapper = shallow( 
            <PrimeraApp 
                saludo={mensaje} 
                subtitulo={ subtitle }
            /> 
        );

        const textoParrafo = wrapper.find('small').text();
        console.log(textoParrafo);

        expect(textoParrafo).toBe(subtitle);
        
    });
    
})