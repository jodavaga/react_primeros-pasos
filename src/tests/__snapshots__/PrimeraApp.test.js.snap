// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primera App tests Should render <PrimeraApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola, soy jose
  </h1>
  <small>
    Subtitulo by default prop
  </small>
</Fragment>
`;
